runtime/src/bank.rs

```diff
@@ -4103,7 +4107,7 @@ impl Bank {
 
             match execution_result.flattened_result() {
                 Ok(()) => {
-                    tx_count += 1;
+                    executed_transactions_count += 1;
```

Comment:
This branch of `flattened_result` means that not only was the tx executed, it returned success. It's been like this for a long time (tx counts are under-reported because the numbers don't include failed txs)

Reply:
I brought this up before awhile ago but I hesitated to fix it because it would modify the bank's tx count. But I think now is a good time to at least fix the metrics even if we keep the bank tx count the same.

Reply:
Fixed here: https://github.com/solana-labs/solana/pull/22802/commits/2e9d821a345912ef69b966c1040226867e7eb48f

Updated the return struct from `Bank::load_and_execute_transactions()` to differentiate `executed_with_successful_result_count` and `executed_transactions_count`, and be mindful to use `executed_with_successful_result_count` in all the same *wrong* places as before, but pass the *correct* executed_transactions_count to callers to setup for correct metrics.